core/crypto/vrf/impl/vrf_provider_impl.hpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_CRYPTO_VRF_IMPL_VRF_PROVIDER_IMPL_HPP
+#define CPP_FILECOIN_CORE_CRYPTO_VRF_IMPL_VRF_PROVIDER_IMPL_HPP
+
+#include "crypto/bls/bls_provider.hpp"
+#include "crypto/vrf/vrf_provider.hpp"
+
+namespace fc::crypto::vrf {
+
+  class VRFProviderImpl : public VRFProvider {
+   public:
+    ~VRFProviderImpl() override = default;
+
+    explicit VRFProviderImpl(std::shared_ptr<bls::BlsProvider> bls_provider);
+
+    outcome::result<VRFResult> computeVRF(
+        const VRFSecretKey &secret_key, const VRFParams &params) const override;
+
+    outcome::result<bool> verifyVRF(const VRFPublicKey &public_key,
+                                    const VRFParams &message,
+                                    const VRFProof &vrf_proof) const override;
+
+   private:
+    std::shared_ptr<bls::BlsProvider> bls_provider_;  ///< bls provider instance
```

Comment:
```suggestion
    std::shared_ptr<bls::BlsProvider> bls_provider_;  ///< bls provider instance
```
This comment doesn't look very meaningful, but if you think keeping it is better for some documentation generation system then ok.

Reply:
probably it is too obvious